Add docs constraints file

Czaki opened the pull request:
# Description
Add compilation of constraints files for docs build. Update workflow to use it. 
 

# References
<!-- What resources, documentation, and guides were used in the creation of this PR? -->

## Type of change
<!-- Please delete options that are not relevant. -->
- [x] Maintenance (changes required to run napari, tests, & CI smoothly)

Czaki committed:
add docs requirements compilation

Czaki committed:
remove docs before create commit

napari-bot committed:
test: [Automatic] Constraints upgrades: `hypothesis` (#5992)

This PR is automatically created and updated by this napari GitHub
Action:

https://github.com/napari/napari/tree/main/.github/workflows/upgrade_test_contraints.yml
It ensures that dependencies are tested with the most recent version.

The updated packages are:

 * hypothesis

Co-authored-by: napari-bot <napari-bot@users.noreply.github.com>

Czaki committed:
use constrians file

Czaki committed:
fix path

psobolewskiPhD commented on .github/workflows/build_docs.yml:
```diff
@@ -60,7 +60,7 @@ jobs:
           # the napari-docs repo is cloned into a docs/ folder, hence the
           # invocation below. Locally, you should simply run make docs
-          run: make -C docs docs GALLERY_PATH=../examples/
+          run:  PIP_CONSTRAINTS=../napari-repo/resources/constraints/constraints_py3.10_docs.txt make -C docs docs GALLERY_PATH=../examples/
```
```suggestion
          run:  PIP_CONSTRAINTS=napari-repo/resources/constraints/constraints_py3.10_docs.txt make -C docs docs GALLERY_PATH=../examples/
```
I think that should fix the build?

Czaki replied:
It looks lie I wrongly understand how `make -C` works.

psobolewskiPhD replied:
The circleCI build where passing `PIP_CONSTRAINTS` isn't erroring, but isn't actually using it, see top lines of:
https://circleci.com/api/v1.1/project/github/napari/napari/786/output/106/0?file=true&allocation-id=64956571b63ae708434971e1-0-build%2F3FC75E41

```
echo clean
clean
echo /home/circleci/project/napari-docs/
/home/circleci/project/napari-docs/
rm -rf /home/circleci/project/napari-docs/docs/_build/
rm -rf /home/circleci/project/napari-docs/docs/api/napari*.rst
rm -rf /home/circleci/project/napari-docs/docs/gallery/*
rm -rf /home/circleci/project/napari-docs/docs/_tags
python -m pip install -qr /home/circleci/project/napari-docs/requirements.txt
python /home/circleci/project/napari-docs/docs/_scripts/prep_docs.py
Cloning into '/home/circleci/project/napari-docs/npe2'...
Warning: Permanently added the ECDSA host key for IP address '140.82.112.4' to the list of known hosts.
```

So I think we either need to change the Makefile or install using pip directly and then `make html` which skips the install.

psobolewskiPhD replied:
If the constraints file (the -docs one) is compiled with napari/docs requirements.txt and is used to install napari[dev] then do we need to use it again when installing the docs? everything should either be installed or compatible...

Czaki replied:
New release of docs dependencies may break env but the probability is lower.

psobolewskiPhD replied:
So maybe we should make a constraints in napari/docs also? and pip install using that and not requirements.txt?

Czaki replied:
In each scenario, we use both repositories. I prefer to update constraints in only one of them.

Carreau committed:
Update .github/workflows/build_docs.yml

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

Czaki committed:
use env variables

psobolewskiPhD commented on .github/workflows/build_docs.yml:
```diff
@@ -57,10 +57,12 @@ jobs:
         env:
           GOOGLE_CALENDAR_ID: ${{ secrets.GOOGLE_CALENDAR_ID }}
           GOOGLE_CALENDAR_API_KEY: ${{ secrets.GOOGLE_CALENDAR_API_KEY }}
+          PIP_CONSTRAINTS: ${{ GITHUB_WORKSPACE }}/napari-repo/resources/constraints/constraints_py3.10_docs.txt
```
```suggestion
          PIP_CONSTRAINTS: ${{ env.GITHUB_WORKSPACE }}/napari-repo/resources/constraints/constraints_py3.10_docs.txt
```

psobolewskiPhD replied:
Not sure why it's not showing, but the build_docs workflow isn't valid:
https://github.com/napari/napari/actions/runs/5356030146

Czaki replied:
it looks working now

Czaki committed:
use github.workspace

Czaki committed:
Merge branch 'main' into docs_constrains

Czaki committed:
fix typo in path

napari-bot committed:
test: [Automatic] Constraints upgrades: `app-model`, `dask`, `hypothesis`, `jsonschema`, `npe2`, `pyyaml`, `tifffile`, `virtualenv`, `xarray`, `zarr` (#6074)

This PR is automatically created and updated by this napari GitHub
Action:

https://github.com/napari/napari/tree/main/.github/workflows/upgrade_test_constraints.yml
It ensures that dependencies are tested with the most recent version.

The updated packages are:

 * app-model
 * click
 * cmake
 * dask
 * distlib
 * fonttools
 * hypothesis
 * jsonschema
 * jsonschema-specifications
 * npe2
 * pip
 * platformdirs
 * pyyaml
 * referencing
 * rpds-py
 * tifffile
 * tomli-w
 * urllib3
 * virtualenv
 * xarray
 * zarr
 * zipp

Co-authored-by: napari-bot <napari-bot@users.noreply.github.com>

Czaki committed:
Merge branch 'main' into docs_constrains

psobolewskiPhD reviewed (approved):
We need to update the napari/docs workflow too right?
In this PR:
https://github.com/napari/docs/pull/213
